src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexProvider.cs

```diff
@@ -59,7 +60,7 @@ internal class FunctionIndexProvider : IFunctionIndexProvider
             _defaultTimeout = timeoutOptions.Value.ToAttribute();
             _allowPartialHostStartup = hostOptions.Value.AllowPartialHostStartup;
             _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-            _defaultRetryStrategy = defaultRetryStrategy;
+            _defaultRetryStrategy = retryOptions.Value.ToAttribute();
```

Comment:
Need a test either unit or e2e (like Timeout_UsingOptions) that verifies we apply options correctly when configured.